lib/java_buildpack/framework/contrast_security_agent.rb

```diff
@@ -78,40 +84,13 @@ def supports?
       private_constant :API_KEY, :FILTER, :INFLECTION_VERSION, :PLUGIN_PACKAGE, :SERVICE_KEY, :TEAMSERVER_URL,
                        :USERNAME
 
-      def add_contrast(doc, credentials)
-        contrast = doc.add_element('contrast')
-        (contrast.add_element 'id').add_text('default')
-        (contrast.add_element 'global-key').add_text(credentials[API_KEY])
-        (contrast.add_element 'url').add_text("#{credentials[TEAMSERVER_URL]}/Contrast/s/")
-        (contrast.add_element 'results-mode').add_text('never')
-
-        add_user contrast, credentials
-        add_plugins contrast
-      end
-
-      def add_plugins(contrast)
-        plugin_group = contrast.add_element('plugins')
-
-        (plugin_group.add_element 'plugin').add_text("#{PLUGIN_PACKAGE}.security.SecurityPlugin")
-        (plugin_group.add_element 'plugin').add_text("#{PLUGIN_PACKAGE}.architecture.ArchitecturePlugin")
-        (plugin_group.add_element 'plugin').add_text("#{PLUGIN_PACKAGE}.appupdater.ApplicationUpdatePlugin")
-        (plugin_group.add_element 'plugin').add_text("#{PLUGIN_PACKAGE}.sitemap.SitemapPlugin")
-        (plugin_group.add_element 'plugin').add_text("#{PLUGIN_PACKAGE}.frameworks.FrameworkSupportPlugin")
-        (plugin_group.add_element 'plugin').add_text("#{PLUGIN_PACKAGE}.http.HttpPlugin")
-      end
-
-      def add_user(contrast, credentials)
-        user = contrast.add_element('user')
-        (user.add_element 'id').add_text(credentials[USERNAME])
-        (user.add_element 'key').add_text(credentials[SERVICE_KEY])
-      end
-
       def application_name
         @application.details['application_name'] || 'ROOT'
       end
 
       def appname_exist?
-        @droplet.java_opts.any? { |java_opt| java_opt =~ /contrast.override.appname/ }
+        @droplet.java_opts.any? { |java_opt| java_opt =~ /contrast.override.appname/ } ||
```

Comment:
You can combine this if you want. it'd save an iteration, but probably doesn't matter. Also probably trivial, but I the `.` here is wildcard and should be `\.`

```ruby
@droplet.java_opts.any? { |java_opt| java_opt =~ /contrast\.override\.appname/ || java_opt =~ /contrast\.application\.name/ }
```